src/python/pants/rules/core/binary.py

```diff
@@ -32,12 +36,26 @@ class CreatedBinary:
 
 
 @console_rule
-async def create_binary(addresses: BuildFileAddresses, console: Console, workspace: Workspace, options: Binary.Options) -> Binary:
+async def create_binary(
+    addresses: BuildFileAddresses,
+    console: Console,
+    workspace: Workspace,
+    options: Binary.Options,
+    options_bootstrapper: OptionsBootstrapper,
+    build_root: BuildRoot
+    ) -> Binary:
   with Binary.line_oriented(options, console) as print_stdout:
-    print_stdout("Generating binaries in `dist/`")
+    global_options = options_bootstrapper.bootstrap_options.for_global_scope()
+    pants_distdir = Path(global_options.pants_distdir)
+    if not is_child_of(pants_distdir, build_root.pathlib_path):
+      console.print_stderr(f"When set to an absolute path, `--pants-distdir` must be relative to the build root. You set it to {pants_distdir}. Instead, use a relative path or an absolute path relative to the build root.")
```

Comment:
This line is more than 100 characters, which is frowned upon. We use implicit string concatenation to get around this, e.g.:

```python
console.print_stderr(
  f"When set to an absolute path, `--pants-distdir` must be relative to the build root. You "
  f"set it to {pants_distdir}. Instead, use a relative path or an absolute path relative to the build root."
)